docs/sources/operations/request-validation-rate-limits.md

```diff
@@ -0,0 +1,233 @@
+
+| Property                | Value      |
+|-------------------------|------------|
+| Enforced by             | `ingester` |
+| Retryable               | **No**     |
+| Sample discarded        | **Yes**    |
+| Configurable per tenant | No         |
+
+## `greater_than_max_sample_age`
+
+If the `reject_old_samples` config option is set to `true` (it is by default), then samples will be rejected with `reason=greater_than_max_sample_age` if they are older than the `reject_old_samples_max_age` value. You should not see samples rejected for `reason=greater_than_max_sample_age` if `reject_old_samples=false`.
+
+This value can be modified globally in the [`limits_config`](https://grafana.com/docs/loki/latest/configuration/#limits_config) block, or on a per-tenant basis in the [runtime overrides](https://grafana.com/docs/loki/latest/configuration/#runtime-configuration-file) file. This error can be solved by increasing the `reject_old_samples_max_age` value, or investigating why log delivery is delayed for this particular stream. The stream in question will be returned in the body of the HTTP response.
+
+| Property                | Value             |
+|-------------------------|-------------------|
+| Enforced by             | `distributor`     |
+| Outcome                 | Request rejected  |
+| Retryable               | **No**            |
+| Sample discarded        | **Yes**           |
+| Configurable per tenant | Yes               |
+| HTTP status code        | `400 Bad Request` |
+
+## `too_far_in_future`
+
+If a sample's timestamp is greater than the current timestamp, Loki allows for a certain grace period during which samples will be accepted. If the grace period is exceeded, the error will occur.
+
+This value can be modified globally in the [`limits_config`](https://grafana.com/docs/loki/latest/configuration/#limits_config) block, or on a per-tenant basis in the [runtime overrides](https://grafana.com/docs/loki/latest/configuration/#runtime-configuration-file) file. This error can be solved by increasing the `creation_grace_period` value, or investigating why this particular stream has a timestamp too far into the future. The stream in question will be returned in the body of the HTTP response.
+
+| Property                | Value             |
+|-------------------------|-------------------|
+| Enforced by             | `distributor`     |
+| Outcome                 | Request rejected  |
+| Retryable               | **No**            |
+| Sample discarded        | **Yes**           |
+| Configurable per tenant | Yes               |
+| HTTP status code        | `400 Bad Request` |
+
+## `max_label_names_per_series`
+
+If a sample is submitted with more labels than Loki has been configured to allow, it will be rejected with the `max_label_names_per_series` reason.
```

Comment:
```suggestion
If a sample is submitted with more labels than Loki has been configured to allow, it will be rejected with the `max_label_names_per_series` reason. Note that 'series' is the same thing as a 'stream' in Loki - the 'series' term is a legacy name. 
```